packages/core/src/command/consolidated/search/fhir-resource/ingest-lexical.ts

```diff
@@ -34,10 +35,8 @@ export async function ingestPatientConsolidated({
 
   const resources =
     bundle.entry?.flatMap(entry => {
-      const resource = entry.resource;
-      if (!resource) return [];
-      if (resource.resourceType === "Patient") return [];
-      return resource;
+      const resource = filterOutPatientAndRemoveDerivedFromExtensions(entry.resource);
+      return resource ?? [];
```

Comment:
style nit, don't think we need to default to flatmap all the time, more straightforward to chain filter and map operations.

const resource = filterOutPatientAndRemoveDerivedFromExtensions(entry.resource);
return resource ?? [];

```
bundle.entry?
  .filter(({ resource }) => resource.resourceType !== "Patient")
  .map(removeDerivedFromExtensions)

export function filterOutPatientAndRemoveDerivedFromExtensions(
  resource?: Resource
): Resource | undefined {
  if (!resource) return undefined;
  if ("extension" in resource && resource.extension) {
    resource.extension = resource.extension.filter(e => !isDerivedFromExtension(e));
  }
  return resource;
}
```

Reply:
Trying to avoid multiple iterations over the list of resources just for code readability - that's why I renamed the function like that, to be super clear what's doing.

Reply:
why? operation chaining / sequentially applying single operations to collections is i (believe) the typescript idiom and is almost never a performance concern. The ideal is code that looks like this

```
bundle.entry?
  .filter(isNotPatient)
  .map(removeDerivedFromExtensions)
```

Are we worried about perf here because of consolidated bundle? Seems like a preemptive optimization to me.

Reply:
Yeah, we can have 10's of thousands records on these arrays. That chaining reads nice, but it essentially duplicates the computation needed to execute it. Totally fine for small arrays, but we're hurting w/ performance and have a significant amount of data to process. Doing this in all places when we can very easily do in single pass doesn't seem the best approach.

Reply:
@leite08 just benchmarked the two implementations on a 1mil resource bundle. 10% worse in runtime perf (250ms), actually _better_ on memory - unless you feel strongly can we do my suggestion? This isn't the bottleneck and I would rather performance concerns not affect how we write and read our code unless it's clearly a problem. We have many function calls in our codebase that can either be "run 3 simple lines of code" or "command that writes to s3 + fires off massive workflow" and there's no way of knowing without looking at the function definition which makes it difficult to trace execution throughout the system.

Reply:
done